File: solutions/go/raindrops/1/raindrops.go
// Package raindrops should have a package comment that summarizes what it's about.
package raindrops

import (
	"strconv"
)

// Convert should have a comment documenting it.
func Convert(n int) string {
	str := ""
	if n%3 == 0 {
		str = "Pling"
	}
	if n%5 == 0 {
		str += "Plang"
	}
	if n%7 == 0 {
		str += "Plong"
	}
	if str == "" {
		str = strconv.Itoa(n)
	}
	return str
}

File: solutions/prolog/raindrops/1/raindrops.pl
convert(N, Sounds) :-
    Pling is N mod 3,
    Plang is N mod 5,
    Plong is N mod 7,
    raindrops(Pling, Plang, Plong, N, Sounds).

raindrops(0, 0, 0, _, Sounds) :- Sounds = "PlingPlangPlong".
raindrops(0, 0, _, _, Sounds) :- Sounds = "PlingPlang".
raindrops(0, _, 0, _, Sounds) :- Sounds = "PlingPlong".
raindrops(_, 0, 0, _, Sounds) :- Sounds = "PlangPlong".
raindrops(0, _, _, _, Sounds) :- Sounds = "Pling".
raindrops(_, 0, _, _, Sounds) :- Sounds = "Plang".
raindrops(_, _, 0, _, Sounds) :- Sounds = "Plong".
raindrops(_, _, _, N, Sounds) :- number_string(N, Sounds).